unlock-app/src/__tests__/middlewares/storageMiddleware.test.js

```diff
@@ -10,6 +14,8 @@ import configure from '../../config'
 import { SIGNUP_CREDENTIALS } from '../../actions/signUp'
 import { LOGIN_CREDENTIALS } from '../../actions/login'
 
+jest.mock('@unlock-protocol/unlock-js')
```

Comment:
this makes me nervous - the test will pass even when the underlying implementation shifts.

Could you add a separate test in a new file that just verifies assumptions about the underlying API? That way we won't blow up when the next version is published

Reply:
alternative: rewrite the test to not mock, and take advantage of `expect.objectContaining` (https://jestjs.io/docs/en/expect#expectobjectcontainingobject) and `expect.any` (https://jestjs.io/docs/en/expect#expectanyconstructor) to ignore the implementation details of what the private key looks like and so on

Reply:
This is like this because when I tried to just spy on `getAccountFromPrivateKey` and it didn't seem to work without doing a whole mock. If you have any suggestions on how to fix that I'm all ears, because I don't like this either

Reply:
Ok, I'll take a look, heading in to the office now

Reply:
No worries, I'll continue to play with it as well